src/main/java/gregtech/api/capability/impl/AbstractRecipeLogic.java

```diff
@@ -258,6 +258,8 @@ protected boolean canProgressRecipe() {
 
         if (requiredType == null) return true;
 
+        if (getMetaTileEntity() instanceof MultiblockWithDisplayBase && ConfigHolder.machines.cleanMultiblocks) return true;
```

Comment:
I would suggest checking `instanceof IMultiblockController` instead here.

Reply:
This line was a direct copy from checkRecipe. Should that one be changed as well?

Reply:
Yes that's probably better to do everywhere in relation to the cleanroom.

Reply:
K. I'll apply it to these two methods for this issue and I can scan some other areas later and do a separate PR if it pops up anywhere else.